version: '3.8'

services:
  historian:
    image: bandung:5000/historian/ai:latest
    environment:
      OPENAI_API_KEY: "${OPENAI_API_KEY}"
      OPENAI_MODEL_NAME: "gpt-4o"
      SERPER_API_KEY: "${SERPER_API_KEY}"
      NEWS_API_KEY: "${NEWS_API_KEY}"
      WEATHER_API_KEY: "${WEATHER_API_KEY}"
    command: streamlit run main.py --server.port=8501 --server.address=0.0.0.0
    deploy:
      replicas: 3
      restart_policy:
        condition: on-failure
      labels:
        - "traefik.enable=true"
        - "traefik.docker.network=traefik-public"
        - "traefik.constraint-label=traefik-public"
        - "traefik.http.middlewares.https-redirect.redirectscheme.scheme=https"
        - "traefik.http.routers.historian.rule=Host(`historian.gaeni.org`)"
        - "traefik.http.routers.historian.entrypoints=https"
        - "traefik.http.routers.historian.tls=true"
        - "traefik.http.routers.historian.tls.certresolver=le"
        - "traefik.http.services.historian.loadbalancer.server.port=8501"
        - "traefik.http.services.historian.loadbalancer.sticky.cookie=true"
    networks:
      - traefik-public

networks:
  traefik-public:
    external: true
